### DISCUSSION TITLE
unable to install netbox_topology_views plugin

### DISCUSSION THREAD

**Discussion by wampfort** [netbox-community/netbox#16249] [Help Wanted!] [answered]
I got stuck with an issue I'm encountering. I installed NetBox version 4.0.1 and tried to install the netbox_topology_views plugin. I followed the simple guide here: https://github.com/netbox-community/netbox-topology-views/tree/develop, but I keep encountering different errors.
At first, I encountered a permission issue while on the virtual environment (venv). To resolve it, I changed the ownership of the /opt/netbox directory to netbox using the command below:
sudo chown -R netbox /opt/netbox
However, this command did not help. What allowed me to proceed was:
sudo chmod -R 757 /opt/netbox/

FYI, I followed the user creation steps from the installation guide:
adduser --system --group netbox

After getting past the permission issue, I encountered another problem saying **no module named extras.plugins.**

I was able to install all the tools listed in the requirements.txt file. After getting exhausted, I tried to restore my snapshot from before the installation, but now I encounter a hash issue and am unable to download it.

![hash](https://github.com/netbox-community/netbox/assets/170566305/d8747cbe-05b4-49be-9a42-043196cec14f)

Thanks for your assistance.

**Comment by jcollie** [comment, marked as answer]
netbox_topology_views has not been updated yet for 4.0. You'll either need to switch to Netbox 3.7 or wait for the plugin to be updated.

https://github.com/netbox-community/netbox-topology-views/pull/495

**Reply by wampfort** [reply]
Oh, thank you so much for your quick response @jcollie . I guess I will wait for it to be supported in our version.

Would you recommend anything for my current setup? I am encountering the permission issue because I don't have a user in the virtual environment (venv) that has permission to modify the /opt/netbox folder.
